src/downloads.jl

```diff
@@ -1,4 +1,19 @@
 import TimeSeries.TimeArray
+
+
+abstract type AbstractDownloadResponse end
+
+struct FredResponse <: AbstractDownloadResponse
+    http_resp::HTTP.Messages.Response
+    m::String
+end
+
+struct YahooResponse <: AbstractDownloadResponse
+    http_resp::HTTP.Messages.Response
+    m::String
+end
```

Comment:
How about making `m` as a type parameter.
So we do not need to double the code.

for example:
```julia
struct APIRespone{T} <: HTTP.Message
  res::HTTP.Response
end

APIResponse{:yahoo}(res)
# and
APIResponse{:fred}(res)
```